lib/npg_qc/autoqc/role/result.pm

```diff
@@ -132,66 +132,71 @@ sub is_old_style_result {
 
 =head2 equals_byvalue
 
-Determines whether the values of the attributes in the object are as listed in the argument hash. 
-Takes a reference to a hash where keys are the names of the attributes and values are expected values. 
-Returns true if all values are as expected, otherwise returns false.
 Supports comparison on the following attributes:
   id_run, position, tag_index, check_name, class_name.
 
- my $r = npg_qc::autoqc::results::result->new({id_run => 222, position => 2, path => q[my_path]});
- $r->equals_byvalue({id_run => 222, position => 2,}); #returns 1
- $r->equals_byvalue({id_run => 222, position => 1,}); #returns 0
- $r->equals_byvalue({id_run => 222,});                #returns 1
+Returns true if this object has the properties listed in a hash.
+
+ my $r = npg_qc::autoqc::results::result->new({id_run => 222, position => 2);
+ $r->equals_byvalue({id_run => 222, position => 2});  # true
+ $r->equals_byvalue({id_run => 222, position => 1});  # false
+ $r->equals_byvalue({id_run => 222});                 # true
+ $r->equals_byvalue({id_run => 222, tag_index => 1}); # false
+ $r->equals_byvalue({check_name => 'result'});        # true
+
+ my $component = npg_tracking::glossary::composition::component::illumina->new(
+   id_run => 1, position => 2);
+ my $f = npg_tracking::glossary::composition::factory->new();
+ $f->add_component($c);
+ my $r = npg_qc::autoqc::results::some->new(
+   composition => $f->create_composition()
+ );
+ $r->equals_byvalue({id_run => 222, position => 1});  # false
+ $r->equals_byvalue({id_run => 1, position => 2});    # true
 
 =cut
 sub equals_byvalue {
   my ($self, $other) = @_;
 
-  if (!defined $other) {
-    croak 'Nothing to compare to';
+  if (!defined $other || ref $other ne 'HASH') {
+    croak 'Can compare to HASH only';
   }
 
-  my $other_type = ref $other;
-  my $comp;
-  if ($other_type) {
-    if ($self->is_old_style_result()) {
-      $comp =  $self->_equals_byvalue_old($other);
-    } else {
-      if ($other_type eq ref $self->composition) {
-        $comp = ($self->composition_digest cmp $other->digest) == 0 ? 1 : 0;
-      }
-    }
-  }
+  my %test = %{$other};
 
-  if (!defined $comp) {
-    croak 'Cannot evaluate input ' . $other;
+  my @attrs = keys %test;
+  if (!@attrs) {
+    croak q[No parameters for comparison];
   }
 
-  return $comp;
-}
-
-sub _equals_byvalue_old {
-  my ($self, $h) = @_;
-
-  my @keys =keys %{$h};
-  if (!@keys) {
-    croak q[No parameters for comparison];
+  for my $attr (qw(class_name check_name)) {
```

Comment:
Do we need the class name comparison? Would we ever be comparing a DB sourced object with a check or json created one (in which case wouldn't the class name be different but check name the same)? 

Anyway, no change in behaviour so should not block this PR, right?